craft_application/util/error_formatting.py

```diff
@@ -70,11 +69,11 @@ def format_pydantic_error(loc: Iterable[str | int], message: str) -> str:
         return f"- duplicate {field_name!r} entry not permitted in {location} configuration"
     if field_path in ("__root__", ""):
         return f"- {message}"
-    return f"- {message} (in field {field_path!r})"
+    return f"- {message} (in field {field_path!r}, input: {value!r})"
 
 
 def format_pydantic_errors(
-    errors: Iterable[error_wrappers.ErrorDict], *, file_name: str = "yaml file"
+    errors: Iterable[ErrorDetails], *, file_name: str = "yaml file"
```

Comment:
I think we have applications using this function directly; is this change api-safe?
https://github.com/canonical/snapcraft/blob/0611c8100107ad7fad478a475c70d1a7b5243981/snapcraft/store/client.py#L30